File: test-programs/verify-index/src/processor.ts
import type { Task, Result, TaskRunner } from "./types.js";
import { parseAndProcess } from "./parser.js";

/**
 * Processor implements TaskRunner.
 * run() calls parseAndProcess().
 */
export class Processor implements TaskRunner {
  async run(task: Task): Promise<Result> {
    return parseAndProcess(task.input);
  }
}

File: test-programs/verify-index-python/src/processor.py
"""Processor implements TaskRunner; run() calls parse_and_process()."""
from .parser import parse_and_process
from .types import Result, Task, TaskRunner


class Processor(TaskRunner):
    """Processor wires parsing and processing together."""

    async def run(self, task: Task) -> Result:
        return parse_and_process(task.input)
